core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -216,7 +216,7 @@ public static ProxyConfiguration load() throws IOException {
      */
     public static URLConnection open(URL url) throws IOException {
         Jenkins h = Jenkins.getInstanceOrNull(); // this code might run on slaves
```

Comment:
My point is that we may not be on the thread that the callable is running in, we may be in a forked thread... hence https://github.com/jenkinsci/jenkins/pull/2100